### PR TITLE
[RAPTOR-5588] Fix R fit bug when handling single-col data

### PR DESCRIPTION
**Pull request by jmbannon**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
If R was passed a dataframe that contained only a target and single column of data, it would coerce X to be a 1-d array which caused problems (the code assumes we're handling a 2-d matrix). This PR ensures the matrix structure isn't dropped by adding DROP=False to all manipulations of X

### CODE REVIEW

**Review by 4dahalibut — APPROVED** [review]

File: tests/conftest.py [hunk 1]
```diff
@@ -179,6 +182,7 @@
     BINARY_TEXT: "binary",
     BINARY_SPACES: "binary",
     REGRESSION: "regression",
+    REGRESSION_SINGLE_COL: "regression",
```
Comment (4dahalibut) [inline tests/conftest.py:185]: why not just use the regression key here...

**Comment by jmbannon** [comment]
jarvis test functional_training_model_templates

**Comment by jmbannon** [comment]
jarvis test cmrunner_integration

**Comment by jmbannon** [comment]
jarvis test cmrunniner_integration

**Review by yakov-g — APPROVED** [review]

**Comment by yakov-g** [comment]
jarvis test cmrunner_integration

**Comment by jmbannon** [comment]
jarvis test cmrunner_integration

**Comment by jmbannon** [comment]
jarvis test functional_training_model_templates

**Pull request merged**